Add explicit support for 'mini.nvim'

echasnovski opened the pull request:
Hi! I would like to add explicit support for [mini.nvim](https://github.com/echasnovski/mini.nvim).

Basic choices are taken from either highlight groups to which 'mini.nvim' makes default links or from analogous plugins.

Differences from default linked groups:
- 'mini.indentscope' is based on 'indent-blankline'.
- 'mini.jump2d' takes color from "SpellRare" to be consistent with 'mini.jump'. Using `nocombine = true` to be consistent (not have italic labels on italic text).
- 'mini.starter' is based on 'dashboard.lua', 'whichkey.lua', 'telescope.lua', and personal choices:
    - `MiniStarterSection` is chosen to be visible (as from "Special" highlight group).
    - `MiniStarterItemBullet` is as border.
    - `MiniStarterItemPrefix` and `MiniStarterQuery` are based on "warning" and "info" diagnostic colors to be opposite of each other.
- 'mini.statusline' is based on 'lualine/themes/kanagawa.lua' and personal choices:
    - All `MiniStatuslineMode*` have bold text as this seems to be used in screenshot.
    - `MiniStatuslineDevinfo` and `MiniStatuslineFileinfo` are "slightly different text".
    - `MiniStatuslineModeOther` is chosen to be different from others.
- 'mini.tabline' is based on explicit 'TabLine*' groups:
    - `MiniTablineCurrent` has bold font to be visually distinctive.
    - `MiniTablineVisible` has slightly darker foreground than `MiniTablineCurrent`.
    - `MiniTablineTabpagesection` is chosen from `Search`.
    - `MiniTablineModified*` groups have inverted `fg` and `bg` of their counterparts.
- 'mini.test' has red for fail and green for pass. Red and green chosen to be visible inside floating window.
- 'mini.trailspace' has group with red background to draw attention.

**Edit**: I've made a second iteration which:
- Resolves conflicts.
- Uses colors attributes only from `theme`.
- Adds explicit support for highlight groups from new modules  that were released in the past two years and for the (hopefully) upcoming 'mini.icons'.

Before:

![kanagawa_before](https://github.com/rebelot/kanagawa.nvim/assets/24854248/6db8c9fc-d291-4846-8731-4f5070109fe6)

After:

![kanagawa_after](https://github.com/rebelot/kanagawa.nvim/assets/24854248/0e736f85-c67d-4522-94aa-6b241c794109)

rebelot reviewed (commented):
please do not use color names directly, instead use their alias defined in themes. This will be safer once other themes will become available.

echasnovski commented:
> please do not use color names directly, instead use their alias defined in themes. This will be safer once other themes will become available.

I replaced "red" and "green" colors with `git.removed` and `git.added`.

The only other places with direct color names are mode highlights for 'mini.statusline'. These are verbatim taken from 'lualine/themes/kanagawa.lua'. What do you prefer to do here? Replace with some theme alias (like `crystalBlue` with `fn`)? There are duplicates there, what should be done then?

echasnovski commented:
Replaced with corresponding theme aliases in 'mini.statusline' highlight groups. I don't really feel that this is optimal because extra attention is needed to have those groups have separate colors.

bluz71 commented:
Is there any reason why this can't be merged?

I have a vested interest in this PR since I want to use the `MiniStatusLine` colors in my own plugin.

Cheers.

echasnovski commented:
@rebelot, I am currently doing second round of adding highlight groups from 'mini.nvim' (after two years of doing the first one). This adds highlight groups from new modules and for the (hopefully) upcoming 'mini.icons' module.

Decided to update this PR instead of opening a new one. I've also resolved conflicts and used only colors from `theme` (no colors directly).

rebelot commented:
I am merging this, but since you are the owner of a pretty vast plug-in ecosystem I really trust you to keep up with possible changes!!
